### DISCUSSION TITLE
Either operate on the solved groundstate(s) OR operate on a self-defined state.

### DISCUSSION THREAD

**Discussion by thefacemyer** [QuSpin/QuSpin#567] [Q&A] [answered]
I have a problem, and I can see two ways of solving it; although, I need help doing either in QuSpin. I want to look at spin excitations at individual chain sites. The two ways I'm thinking are:

1. Having numerically calculated the groundstate, how can I tell QuSpin to operate on one of the sites with say S+?
2. How can I create my own state in QuSpin, a linear combination of Fock states with specified cofactors, so that I can calculate expectation values with it?

**Comment by mgbukov** [comment]
1. you can define S+_j the same way you define any other operator, just restrict the support to the site j. What you should watch out for, is that the basis you computed the GS with may not be compatible with constructing the operator S+_j: what I mean here is this: S+_j is not translation invariant (nor parity/reflection invariant) and quspin will throw an error. If that's the case, you should construct S+_j in a basis without symmetries, and transform your GS to the same basis (see, e.g., `basis.project_from()`) in order to be able to operate on the GS. 

2. see [this example](https://weinbe58.github.io/QuSpin/examples/example00.html#example-00-label).

**Reply by thefacemyer** [reply]
It throws an error because applying "+" contain non-hermitian couplings. I'm trying my hand at this with an L=2, S=1/2 chain to make sure the operation will give me correct on-site spin-flipping. (CAVEAT: I'm relatively new at programming.)

**Comment by thefacemyer** [comment, marked as answer]
I believe it worked by turning off the hermicity and symmetry checks. At least it gave me the correct state when I used L=2, S=1/2 as a test.
